docs/src/pages/components/data-grid/sorting/OrderSortingPerColumnGrid.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import { GridRowsProp, DataGrid } from '@mui/x-data-grid';
+import { randomQuantity } from '@mui/x-data-grid-generator';
+
+const rows: GridRowsProp = [
+  {
+    id: 1,
+    platform: 'win-x86',
+    downloads: randomQuantity(),
+  },
+  {
+    id: 2,
+    platform: 'win-x64',
+    downloads: randomQuantity(),
+  },
+  {
+    id: 3,
+    platform: 'win-arm',
+    downloads: randomQuantity(),
+  },
+  {
+    id: 4,
+    platform: 'win-arm64',
+    downloads: randomQuantity(),
+  },
+  {
+    id: 5,
+    platform: 'linux-x64',
+    downloads: randomQuantity(),
+  },
+  {
+    id: 6,
+    platform: 'linux-arm',
+    downloads: randomQuantity(),
+  },
+  {
+    id: 7,
+    platform: 'linux-arm64',
+    downloads: randomQuantity(),
+  },
+  {
+    id: 8,
+    platform: 'osx-x64',
+    downloads: randomQuantity(),
+  },
+];
+
+export default function OrderSortingPerColumnGrid() {
+  return (
+    <div style={{ height: 400, width: '100%' }}>
```

Comment:
Could you use `useDemoData` here, instead of hardcoding the rows and columns? The quantity column could be used instead of the downloads.

Reply:
Updated demo code (not sure how readable this is):
```tsx
<DataGrid
  {...data}
  columns={data.columns.map((column) => ({
    ...column,
    sortingOrder:
      column.field === 'quantity' ? ['desc', 'asc', null] : undefined,
  }))}
  sortingOrder={['asc', 'desc', null]}
/>
```